pkg/backup/restore/restore_manager.go

```diff
@@ -236,8 +247,70 @@ func (rm *restoreManager) readRestoreMetaFromExternalStorage(r *v1alpha1.Restore
 
 	return csb, "", nil
 }
+func (rm *restoreManager) validateRestore(r *v1alpha1.Restore, tc *v1alpha1.TidbCluster) error {
+	// check tiflash replicas
+	replicas, reason, err := rm.readTiFlashReplicasFromBackupMeta(r)
+	if err != nil {
+		klog.Errorf("read tiflash replica failure with reason %s", reason)
+		return err
+	}
+
+	if tc.Spec.TiFlash == nil {
+		if replicas != 0 {
+			klog.Errorf("tiflash is not configured, backupmeta has %d tiflash", replicas)
+			return fmt.Errorf("tiflash replica missmatched")
+		}
+
+	} else {
+		if tc.Spec.TiFlash.Replicas != replicas {
+			klog.Errorf("cluster has %d tiflash configured, backupmeta has %d tiflash", tc.Spec.TiFlash.Replicas, replicas)
+			return fmt.Errorf("tiflash replica missmatched")
+		}
+	}
+
+	return nil
+}
+
+func (rm *restoreManager) readTiFlashReplicasFromBackupMeta(r *v1alpha1.Restore) (int32, string, error) {
+	// since the restore meta is small (~5M), assume 1 minutes is enough
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(time.Minute*1))
+	defer cancel()
+
+	klog.Infof("read the backup meta from external storage")
+	cred := backuputil.GetStorageCredential(r.Namespace, r.Spec.StorageProvider, rm.deps.SecretLister)
+	externalStorage, err := backuputil.NewStorageBackend(r.Spec.StorageProvider, cred)
+	if err != nil {
+		return 0, "NewStorageBackendFailed", err
+	}
+
+	// if file doesn't exist, br create volume has problem
+	exist, err := externalStorage.Exists(ctx, constants.MetaFile)
+	if err != nil {
+		return 0, "FileExistedInExternalStorageFailed", err
+	}
+	if !exist {
+		return 0, "FileNotExists", fmt.Errorf("%s does not exist", constants.MetaFile)
+	}
+
+	restoreMeta, err := externalStorage.ReadAll(ctx, constants.MetaFile)
+	if err != nil {
+		return 0, "ReadAllOnExternalStorageFailed", err
+	}
+
+	metaInfo := &backuputil.EBSBasedBRMeta{}
+	err = json.Unmarshal(restoreMeta, metaInfo)
```

Comment:
Do we have a common method to get restore meta from external storage? If not exist, I suggest to split it to a common method.

Reply:
fixed in https://github.com/pingcap/tidb-operator/pull/4812/commits/30ac10ba55a7b72b5780652c5163c320d99a4453